src/OpenApi/Factory/OpenApiFactory.php

```diff
@@ -265,7 +265,14 @@ private function collectPaths(ApiResource $resource, ResourceMetadataCollection
                     continue;
                 }
 
-                $parameter = new Parameter($parameterName, 'path', (new \ReflectionClass($uriVariable->getFromClass()))->getShortName().' identifier', true, false, false, ['type' => 'string']);
+                $fromClass = $this->resourceMetadataFactory->create($uriVariable->getFromClass());
```

Comment:
interesting but I think that this is quite a lot of work, probably that this should actually get fixed by #5995